src/indexer/index.ts

```diff
@@ -11,3 +17,87 @@ export const searchForAssets = async (tokenName: string) => {
 		console.log(e);
 	}
 };
+
+export const searchForApplication = async (application_id: number) => {
+	try {
+		const applicationInfo = await indexerClient
+			.lookupApplications(application_id)
+			.do();
+
+		const creator = applicationInfo.application.params.creator;
+		const globalState = await readAppGlobalState(creator, application_id);
+		return globalState;
+	} catch (e) {
+		console.log(e);
+		throw e;
+	}
+};
+
+export const searchForAccount = async (
+	address: string,
+	application_id: number
+) => {
+	try {
+		const accountInfo = await indexerClient.lookupAccountByID(address).do();
+		console.log(
+			"Information for Account: " + JSON.stringify(accountInfo, undefined, 2)
+		);
+		const localStateMap = await readAppLocalState(address, application_id);
+		const total_amount = accountInfo.account.amount;
+		return { total_amount, localStateMap };
+	} catch (e) {
+		console.log(e);
+		throw e;
+	}
+};
+
+export async function readAppGlobalState(
+	creator: AccountAddress,
+	appID: number
+): Promise<Map<Key, StateValue> | undefined> {
+	const accountInfoResponse = await algodClient
+		.accountInformation(creator)
+		.do();
+	for (const app of accountInfoResponse["created-apps"]) {
+		if (app.id === appID) {
+			const globalStateMap = new Map<Key, StateValue>();
+			for (const g of app.params["global-state"]) {
+				const key = Buffer.from(g.key, "base64").toString();
+				if (g.value.type === 1) {
+					globalStateMap.set(
+						key,
+						Buffer.from(g.value.bytes, "base64").toString("ascii")
+					);
+				} else {
+					globalStateMap.set(key, g.value.uint);
+				}
+			}
+			return globalStateMap;
+		}
+	}
+	return undefined;
+}
+
+export async function readAppLocalState(
+	account: AccountAddress,
+	appID: number
+): Promise<Map<Key, StateValue> | undefined> {
+	const accountInfoResponse = await algodClient
+		.accountInformation(account)
+		.do();
+	for (const app of accountInfoResponse["apps-local-state"]) {
+		if (app.id === appID) {
+			const localStateMap = new Map<Key, StateValue>();
```

Comment:
I see we share same logic when create {local,global}StateMap. Could we create function for it?

Reply:
yes that's right just the keys are different, will update